File: eureka/ios/nativeLib/nativeLib/Models/ACharDef.swift
//
//  ACharDef.swift
//  BLEDemo
//

import Foundation

public class ACharDef {
    public final var uuid: UUID?
    public final var charEnum: GattCharEnum
    public final var name: String
    public final var dataType: DataTypeEnum
    public final var property: CharPropertyEnum
    public final var length: Int
    public final var members: [AMember] = []
    
    public init(pUuid: String, pCharEnum: GattCharEnum, pName: String, pDataType: DataTypeEnum, pProperty:CharPropertyEnum, pLength: Int) {
        uuid = UUID(uuidString: pUuid)
        charEnum = pCharEnum
        name = pName
        dataType = pDataType
        property = pProperty
        length = pLength
    }
    
    public func addMember(pMember: AMember) {
        members.append(pMember)
    }
    
    public func addMember(pName: String, pDataType: DataTypeEnum, pLength: Int) {
        members.append(AMember(pName: pName, pDataType: pDataType, pLength: pLength));
    }
    
    public func getUuid() -> UUID? {
        return uuid
    }
    
    public func getUuidStr() -> String {
        return uuid?.description ?? ""
    }
    
    public func getMembers() -> [AMember] {
        return members
    }
}

File: eureka/android/library/src/main/java/com/lifeplus/Pojo/ACharDef.java
package com.lifeplus.Pojo;

import com.lifeplus.Pojo.Enum.CharPropertyEnum;
import com.lifeplus.Pojo.Enum.DataTypeEnum;
import com.lifeplus.Pojo.Enum.GattCharEnum;

import java.util.ArrayList;
import java.util.UUID;

public class ACharDef {
    public final UUID uuid;
    public final GattCharEnum charEnum;
    public final String name;
    public final DataTypeEnum dataType;
    public final CharPropertyEnum property;
    public final int length;
    public final ArrayList<AMember> members = new ArrayList<>();

    public ACharDef(String pUuid, String pName, DataTypeEnum pDataType, int pLength, CharPropertyEnum pProperty, GattCharEnum pCharEnum) {
        uuid = UUID.fromString(pUuid);
        charEnum = pCharEnum;
        name = pName;
        dataType = pDataType;
        length = pLength;
        property = pProperty;
    }

    public void addMember(AMember pMember) {
        members.add(pMember);
    }

    public void addMember(String pName, DataTypeEnum pDataType, int pLength) {
        members.add(new AMember(pName, pDataType, pLength));
    }

    public UUID getUuid() {
        return uuid;
    }

    public String getUuidStr() {
        return (uuid != null) ? uuid.toString() : "";
    }

    public ArrayList<AMember> getMembers() {
        return members;
    }
}
